Sources/TuistGenerator/Linter/ProjectLinter.swift

```diff
@@ -50,7 +50,40 @@ class ProjectLinter: ProjectLinting {
         return try await project.targets.values
             .map { $0 }
             .concurrentFlatMap { target in
-                try await self.targetLinter.lint(target: target, options: project.options)
+                return switch project.type {
+                case .local:
+                    // only do source file lint for local target, or for some external target like systemLibrary, it always raise
+                    // warning
+                    try await self.targetLinter.lint(target: target, options: project.options)
+                        + self.lintHasSourceFiles(target: target)
+                case .external:
+                    try await self.targetLinter.lint(target: target, options: project.options)
+                }
```

Comment:
Is the distinction between `external` and `local` why `lintHasSourceFiles` was moved to a `ProjectLinter`?

Is there a reason why this linting could have stayed in the `TargetLinter`? You can use the `Target.type` property that has `local` and `remote` cases that have the same meaning.

Reply:
restored this change and skipped source file lint for remote target